sbt-crossproject/src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -208,26 +209,38 @@ object CrossProject {
     private def sharedSrcSettings(crossType: CrossType): Seq[Setting[_]] = {
       def makeCrossSources(sharedSrcDir: Option[File],
                            scalaBinaryVersion: String,
+                           scalaEpochVersion: String,
                            cross: Boolean): Seq[File] = {
         sharedSrcDir match {
           case Some(dir) =>
             if (cross)
-              Seq(dir.getParentFile / s"${dir.name}-$scalaBinaryVersion", dir)
+              Seq(
+                dir.getParentFile / s"${dir.name}-$scalaBinaryVersion",
+                dir.getParentFile / s"${dir.name}-$scalaEpochVersion",
+                dir
+              )
             else
               Seq(dir)
           case None => Seq()
         }
       }
 
+      def scalaEpochVersion(scalaBinaryVersion: String) = {
+        val isDotty = ScalaInstance.isDotty(scalaBinaryVersion)
+        if (isDotty) "3" else "2"
```

Comment:
This is really just `scalaBinaryVersion.takeWhile(_ != '.')`. There is no need for `ScalaInstance.isDotty`, nor to special-case 2 and 3.

Reply:
I completely agree. I took this code straight out of sbt and didn't ask questions about why they do it like this.